src/Request.js

```diff
@@ -245,17 +246,23 @@ export default class Request {
 
     retry () {
 
-        if (this.retries++ < this._max_retry) {
+        if (this.last_error) {
+            this.errors.push(this.last_error);
+        }
+
+        if (this.retries + 1 < this._max_retry) {
```

Comment:
if we use `++this.retries`, the current count of retries will keep incrementing. let's say it can go up to 10/3 (10 retries out of 3 max, 7 are all max retries reached error). if I want to add 3 more retries, I'll have to set the max_retry to 13.

if we only increment inside the loop, the retry will not increment over the max retry count. 3/3 even if it was retried 10 times. now, if they want to add 3 more retries, they just have to set max_retry to 6.